### DISCUSSION TITLE
Credential provider:  session question (user object is different)

### DISCUSSION THREAD

**Discussion by madeinspace** [nextauthjs/next-auth#2853] [Help]
howdy,

When I authenticate a user using the authorize method from the Credentials provider the user object that I'm returning ({ id: 214, name: 'darebin' } ) is different in the session ie this is what I get when I console out both user and session in the session callback :

```
  user:  { name: 'darebin', sub: '214', iat: 1632818784, exp: 1635410784 }
  session:  {
    user: { name: 'darebin', email: null, image: null },
    expires: '2021-10-28T08:46:27.387Z'
  }
```

Where does this email:null, image:null come from? why hasn't it been replaced by the user object I returned ({ id: 214, name: 'darebin' } )

**Comment by balazsorban44** [comment]
https://next-auth.js.org/configuration/callbacks

**Reply by madeinspace** [reply]
Oh ok so the user has to be overridden in the sign-in callback? I'll try that thank you.

**Reply by balazsorban44** [reply]
Not the sign-in callback. That is used to stop the sign-in process if you have some custom conditions. Check the `jwt` and `session` callbacks.
